src/util/os_utils.js

```diff
@@ -341,12 +340,62 @@ function get_disk_mount_points() {
         });
 }
 
+function _df_parse(stdout) {
+    return stdout
+            .trim()
+            .split(/\r\n|\r|\n/) // split into rows
+            .slice(1) // strip column headers away
+            .map(function(row) {
+                const columns = row
+                // one or more whitespaces followed by one or more digits
+                // must be interpreted as column delimiter
+                .replace(/\s+(\d+)/g, '\t$1')
+                // one or more whitespaces followed by a slash
+                // must be interpreted as the last column delimiter
+                .replace(/\s+\//g, '\t/')
+                // split into columns
+                .split(/\t/);
+
+        return {
+            filesystem: columns[0],
+            size: columns[1],
+            used: columns[2],
+            available: columns[3],
+            capacity: parseInt(columns[4], 10) * 0.01,
+            mount: columns.slice(5).join(' '),
+        };
+    });
+}
+
+async function _df(file_path, callback, flags = '') {
+    const file_name_regex = /[^-_./a-zA-Z0-9]/g;
+    const flags_argument = '-kP' + flags;
+    const args = [flags_argument];
+    if (file_path) {
+        // CVE-2019-15597 add '\' to special charecters to prevent command injection
+        file_path = file_path.replace(file_name_regex, match => '\\' + match);
+        args.push(file_path);
+    }
+    child_process.execFile('/bin/df', args, function(err, stdout, stderr) {
+        if (err) {
+            dbg.error("df command failed:", err);
+            callback(err, null);
+            return;
+        }
+        if (stderr) {
+            dbg.error("df command failed:", stderr);
+            callback(new Error(stderr), null);
+            return;
+        }
+        const drives = _df_parse(stdout);
+        callback(null, drives);
+
+     });
+}
```

Comment:
Not sure we should use callbacks anymore, maybe use promisify? 
```suggestion
async function _df(file_path, flags = '') {
    const file_name_regex = /[^-_./a-zA-Z0-9]/g;
    const flags_argument = '-kP' + flags;
    const args = [flags_argument];

    if (file_path) {
        // CVE-2019-15597 add '\' to special characters to prevent command injection
        file_path = file_path.replace(file_name_regex, match => '\\' + match);
        args.push(file_path);
    }

    try {
        const execFile = util.promisify(child_process.execFile);
        const { stdout, stderr } = await execFile('/bin/df', args);
        if (stderr) {
            throw new Error(stderr);
        }
        return _df_parse(stdout);
    } catch (err) {
        console.error('df command failed:', err);
        throw err;
    }
}
```